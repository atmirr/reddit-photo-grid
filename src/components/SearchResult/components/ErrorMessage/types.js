// @flow
export type ErrorMessageProps = {
  message: string,
};
