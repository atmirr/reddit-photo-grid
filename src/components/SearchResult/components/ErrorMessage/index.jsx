// @flow
import React from 'react';
import type { Element } from 'react';
import styled from 'styled-components';
import type { ErrorMessageProps } from './types.js';

const ErrorText = styled.p`
  font-size: 1rem;
  text-align: center;
  line-height: 1.5em;
  color: #d43b3b;
`;

function ErrorMessage({ message }: ErrorMessageProps): Element<'div'> {
  return <ErrorText>{message}</ErrorText>;
}

export default ErrorMessage;
