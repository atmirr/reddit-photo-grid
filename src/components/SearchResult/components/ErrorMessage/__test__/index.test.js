import React from 'react';
import { render } from '@testing-library/react';
import ErrorMessage from '../index';

describe('<ErrorMessage />', () => {
  it('should renders the component', () => {
    const { container } = render(<ErrorMessage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display the error message when we set message prop', () => {
    const error = 'Error: This keyword is forbidden!';
    const { queryByText } = render(<ErrorMessage message={error} />);
    expect(queryByText(error)).toBeVisible();
  });
});
