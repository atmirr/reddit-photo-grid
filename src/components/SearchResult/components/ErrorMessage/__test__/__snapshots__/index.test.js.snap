// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should renders the component 1`] = `
<p
  class="sc-bdfBwQ gjzUfc"
/>
`;
